github/provider.go

```diff
@@ -217,7 +218,7 @@ func providerConfigure(p *schema.Provider) schema.ConfigureFunc {
 			}
 
 			if v, ok := appAuthAttr["pem_file"].(string); ok && v != "" {
-				appPemFile = v
+				appPemFile = strings.Replace(v, `\n`, "\n", -1)
```

Comment:
Do you think it's worth writing a comment here to explain this behavior? I think it's not super obvious and probably worth documenting in some manner.

Reply:
I have added a comment to the code with some context. 

I chose to add new line replacement in the provider's configuration code because it felt most appropriate as it is dealing with how the provider is configured, however, it could be more clear if the behaviour was added to the part of the provider where the PEM is consumed, i.e: https://github.com/integrations/terraform-provider-github/blob/main/github/apps.go#L19

```diff
func GenerateOAuthTokenFromApp(baseURL, appID, appInstallationID, pemData string) (string, error) {
+       pemData = strings.Replace(pemData, `\n`, "\n", -1)
	appJWT, err := generateAppJWT(appID, time.Now(), []byte(pemData))
```

_Personally_ I think it belongs in the configuration code but I am happy to move it into the consuming code (either [`apps.go@GenerateOAuthTokenFromApp`](https://github.com/integrations/terraform-provider-github/blob/main/github/apps.go#L19) or [`apps.go@generateAppJWT`](https://github.com/integrations/terraform-provider-github/blob/main/github/apps.go#L75)) if you think that will be clearer about the behaviour and intent.

Reply:
Thanks for doing that! I agree with you that this change belongs at the configuration level, so all consumers can trust they have the correctly-escaped version of the certificate. 

I was a little concerned that this might break existing certiricate handling somehow, but I believe that the `base64`-encoded nature of the certificates guarantees that there won't be `\n` string literals in the cert that we accidentally over-replace, so I think this is pretty safe.